internal/awsservice/cloudwatchlogs.go

```diff
@@ -134,30 +72,50 @@ func DeleteLogGroup(logGroupName string) {
 	}
 }
 
-// ValidateLogsInOrder takes a log group, log stream, a list of specific log lines and a timestamp.
-// It should query the given log stream for log events, and then confirm that the log lines that are
-// returned match the expected log lines. This also sanitizes the log lines from both the output and
-// the expected lines input to ensure that they don't diverge in JSON representation (" vs ')
-func ValidateLogsInOrder(t *testing.T, logGroup, logStream string, logLines []string, since time.Time) {
-	log.Printf("Checking %s/%s since %s for %d expected logs", logGroup, logStream, since.UTC().Format(time.RFC3339), len(logLines))
-	cwlClient, clientContext, err := getCloudWatchLogsClient()
+// ValidateLogs queries a given LogGroup/LogStream combination given the start and end times, and executes an
+// arbitrary validator function on the found logs.
+func ValidateLogs(logGroup, logStream string, since, until *time.Time, validator func(logs []string) bool) (bool, error) {
+	log.Printf("Checking %s/%s\n", logGroup, logStream)
+
```

Comment:
I know this is an existing code, but it makes no sense why validation is in this package. This package seems like a SAO.

Reply:
I wrote this code originally with the intent of obfuscating the SDK client and just having validation functions that implicitly used them. I wanted to simplify the testing so that it was easy to write a new test that uses a validation function which makes calls to AWS, without getting into the business of passing around SDK clients. 

Obviously, things got shifted around and that doesn't make sense anymore. I can rip it out now, or can do it in another PR. Up to you